ansible/roles/test/files/tools/loganalyzer/loganalyzer_analyze.yml

```diff
@@ -35,12 +35,12 @@
 
 - name: Add common match file
   set_fact:
-      match_file_option: "{{ match_file_option }}{{ match_file }},"
+      match_file_option: "{{ match_file_option }}{{ match_file }}"
   when: skip_common_match is not defined
 
 - name: Add test specific match file
   set_fact:
-      match_file_option: "{{ match_file_option }}{{ test_match_file }}"
+      match_file_option: "{{ match_file_option }},{{ test_match_file }}"
```

Comment:
If variable "skip_common_match" will be defined, this approach will not work as well.
I suggest to replace "Add test specific match file" with:

- name: Add test specific match file
  set_fact:
      match_file_option: "{{ match_file_option }}{{ test_match_file }}"
  when: test_match_file is defined and skip_common_match is defined

- name: Add test specific match file
  set_fact:
      match_file_option: "{{ match_file_option }},{{ test_match_file }}"
  when: test_match_file is defined and skip_common_match is not defined

Reply:
Clarification: add both tasks.

Reply:
Thanks for your comments!

I pushed a new commit and used a different logic which I think is more clear:

```
- name: Collect list of match files - common_match
  set_fact:
      match_file_list: "{{ match_file_list | default([]) }} + {{ [match_file] }}"
  when: skip_common_match is not defined

- name: Collect list of match files - test_match
  set_fact:
      match_file_list: "{{ match_file_list | default([]) }} + {{ [test_match_file] }}"
  when: test_match_file is defined

- name: Add mmatch file option
  set_fact:
      match_file_option: "{{ match_file_option }}{{ match_file_list | join(',') }}"
```